### PR TITLE
Fix azure test cases with puppet_enabled sat[API & CLI]

### PR DESCRIPTION
**Pull request by adarshdubey-star**
Fixing compute resources - Azure tests to use the puppet-enabled SAT.
For better results, run with this nailgun PR https://github.com/SatelliteQE/nailgun/pull/823

**Test Results**

```
(workenv) [addubey@addubey robottelo]$ pytest tests/foreman/api/test_computeresource_azurerm.py
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.8.6, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/addubey/work/robottelo, configfile: pyproject.toml
plugins: reportportal-5.0.12, services-2.2.1, mock-3.7.0, forked-1.3.0, ibutsu-2.0.2, cov-3.0.0, xdist-2.5.0
collected 6 items                                                                                                                                                                                                 

tests/foreman/api/test_computeresource_azurerm.py ......
```

### CODE REVIEW

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_azurerm.py
nailgun: 823

File: pytest_fixtures/component/provision_azure.py [hunk 1]
```diff
@@ -26,58 +25,66 @@ def azurerm_settings():
 
 
 @pytest.fixture(scope='module')
-def module_azurerm_cr(azurerm_settings, module_org, module_location):
+def module_azurerm_cr(
+    azurerm_settings, module_puppet_org, module_puppet_loc, session_puppet_enabled_sat
```
[the remark this answers is not in the document]
Comment (adarshdubey-star) [reply pytest_fixtures/component/provision_azure.py:29]: We have a fixture in puppet.py as mentioned below for Org/location is this the thing you are talking about ?

@pytest.fixture(scope='module')
def module_puppet_org(session_puppet_enabled_sat):
    yield session_puppet_enabled_sat.api.Organization().create()

**Review by vsedmik — COMMENTED** [review]
Pending question about the `module_azurerm_cr`.

Btw, how did you run the tests? Was it including checkout+installer run by the fixture, or with an existing pre-enabled sat mocked as `session_puppet_enabled_sat`? I use the later for faster debug but in the end I run the first option at least once.

File: pytest_fixtures/component/provision_azure.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (JacobCallahan) [reply pytest_fixtures/component/provision_azure.py:29]: @adarshdubey-star hmmm that is a bit nonobvious, but we don't currently have a policy for noting where external fixtures come from, so I won't ask that here

**Review by JacobCallahan — APPROVED** [review]
ACK. Merge pending other comments

File: pytest_fixtures/component/provision_azure.py [hunk 2]
```diff
@@ -26,58 +25,66 @@ def azurerm_settings():
...
+        organization=[module_puppet_org],
+        location=[module_puppet_loc],
     ).create()
     return azure_cr
 
 
 @pytest.fixture(scope='module')
-def module_azurerm_finishimg(default_architecture, default_os, module_azurerm_cr):
+def module_azurerm_finishimg(
+    default_architecture, session_puppet_default_os, session_puppet_enabled_sat, module_azurerm_cr
```
[the remark this answers is not in the document]
Comment (adarshdubey-star) [reply pytest_fixtures/component/provision_azure.py]: Thanks for your input @vsedmik I have updated the architecture parameter to use the `puppet enabled sat`.

**Review by tstrych — APPROVED** [review]
LGTM. Minor change requested. Please provide test results before this is merged.

File: pytest_fixtures/component/provision_azure.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mshriver) [reply pytest_fixtures/component/provision_azure.py:29]: That's what `pytest --setup-plan` is for!

**Comment by jyejare** [comment]
@adarshdubey-star Does tests on 6.11 executed and passed without any `azure_client` failures? Asking cause 6.10 tests are failing with azure client not recognizing the provisioned VM immediately and if in 7.0 its recognizing then I assume its a bug in 6.10 !

**Review by vsedmik — APPROVED** [review]

**Comment by adarshdubey-star** [comment]
> @adarshdubey-star Does tests on 6.11 executed and passed without any `azure_client` failures? Asking cause 6.10 tests are failing with azure client not recognizing the provisioned VM immediately and if in 7.0 its recognizing then I assume its a bug in 6.10 !

Yes @jyejare they were passing on 7.0 , I tried a re-run and I am confused now, how they are failing. :(

**Comment by adarshdubey-star** [comment]
The cause of the test failures was the absence of the resources at the Azure's end @jyejare restored some of our resources.
Also, I have updated the PR with ```azure_cli``` fix as well as some changes in the fixtures so the non-puppet related tests aren't hampered.

**Updates test results**

 pytest  tests/foreman/cli/test_computeresource_azurerm.py
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.8.6, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/addubey/work/robottelo, configfile: pyproject.toml
plugins: reportportal-5.0.12, services-2.2.1, mock-3.7.0, forked-1.3.0, ibutsu-2.0.2, cov-3.0.0, xdist-2.5.0
collected 3 items                                                                                                                                                                                                 

tests/foreman/cli/test_computeresource_azurerm.py ...                                                                                                                                                       [100%]

=================================================================================== 3 passed, 54 warnings in 928.00s (0:15:28) ====================================================================================



pytest  tests/foreman/api/test_computeresource_azurerm.py -sv
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.8.6, pytest-6.2.5, py-1.10.0, pluggy-1.0.0 -- /home/addubey/work/workenv/bin/python3.8
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/addubey/work/robottelo, configfile: pyproject.toml
plugins: reportportal-5.0.12, services-2.2.1, mock-3.7.0, forked-1.3.0, ibutsu-2.0.2, cov-3.0.0, xdist-2.5.0
collecting ... 2022-04-05 16:50:29 - robottelo.collection - INFO - Processing test items to add testimony token markers
collected 6 items 


tests/foreman/api/test_computeresource_azurerm.py::TestAzureRMHostProvisioningTestCase::test_positive_azurerm_host_provisioned PASSED
tests/foreman/api/test_computeresource_azurerm.py::TestAzureRMHostProvisioningTestCase::test_positive_azurerm_host_power_on_off PASSED
tests/foreman/api/test_computeresource_azurerm.py::TestAzureRMUserDataProvisioning::test_positive_azurerm_ud_host_provisioned PASSED
tests/foreman/api/test_computeresource_azurerm.py::TestAzureRMUserDataProvisioning::test_positive_host_disassociate_associate PASSED
tests/foreman/api/test_computeresource_azurerm.py::TestAzureRMCustomImageFinishTemplateProvisioning::test_positive_azurerm_custom_image_host_provisioned ERROR
tests/foreman/api/test_computeresource_azurerm.py::TestAzureRMSharedGalleryFinishTemplateProvisioning::test_positive_azurerm_shared_gallery_host_provisioned ERROR

============================================================================== 4 passed, 55 warnings, 2 errors in 969.38s (0:16:09) ==============================================================================

Last two cases are failing because of absence of  image_resources.

**Comment by adarshdubey-star** [comment]
We are now good with the last two cases as well.

**Test Results**

pytest  tests/foreman/api/test_computeresource_azurerm.py::TestAzureRMSharedGalleryFinishTemplateProvisioning
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.8.6, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/addubey/work/robottelo, configfile: pyproject.toml
plugins: reportportal-5.0.12, services-2.2.1, mock-3.7.0, forked-1.3.0, ibutsu-2.0.2, cov-3.0.0, xdist-2.5.0
collected 1 item                                                                                                                                                                                                  

tests/foreman/api/test_computeresource_azurerm.py .                                                                                                                                                         [100%]

==================================================================================== 1 passed, 7 warnings in 598.86s (0:09:58) ====================================================================================

 pytest  tests/foreman/api/test_computeresource_azurerm.py::TestAzureRMCustomImageFinishTemplateProvisioning
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.8.6, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/addubey/work/robottelo, configfile: pyproject.toml
plugins: reportportal-5.0.12, services-2.2.1, mock-3.7.0, forked-1.3.0, ibutsu-2.0.2, cov-3.0.0, xdist-2.5.0
collected 1 item                                                                                                                                                                                                  

tests/foreman/api/test_computeresource_azurerm.py .                                                                                                                                                         [100%]

==================================================================================== 1 passed, 7 warnings in 582.90s (0:09:42) ====================================================================================

**Review by jyejare — APPROVED** [review]

**Review by JacobCallahan — APPROVED** [review]

**Pull request merged**